packages/plugins/lib/Plugin.ts

```diff
@@ -58,6 +58,24 @@ export class Plugin {
     return this.loadModule(this.definition.preserve.recipe).Recipe;
   }
 
+  /*
+   * compiler
+   */
+
+  definesCompiler(): boolean {
+    return !!this.definition.compile;
+  }
+
+  loadCompiler(): any {
+    if (!this.definesCompiler()) {
+      throw new TruffleError(
+        `Plugin ${this.module} does not define a \`truffle plugin\` compiler.`
+      );
+    }
+
+    return this.loadModule(this.definition.compile).Compile;
```

Comment:
```suggestion
    return this.loadModule(this.definition.compiler).Compiler;
```

Reply:
@dkillen thanks for reviewing the PR.

1. `compile` in definition.compile is used because `compile` is defined in truffle-plugin.json.  @gnidan 
and @fainashalts can we use `compiler` instead of `compile` as definition in truffle-plugin.json?

2. `Compile` in ().Compile is used because Truffle compilers use `Compile` as the name of API's, e.g., see packages/compile-solidity/src/index.js.  The plugin needs to be consistent with Truffle implementation.  Hence, we cannot change it.

Reply:
Per the discussion in the config meeting, we will change `compile` to `compiler` in the definition.